tbx/services/blocks.py

```diff
@@ -38,8 +40,44 @@ class Meta:
         template = "patterns/molecules/streamfield/blocks/partners_block.html"
 
 
-class ServiceStoryBlock(StoryBlock):
+class CaseStudyStructValue(blocks.StructValue):
+    @cached_property
+    def featured_case_study_image(self):
+        if image := self.get("image"):
+            return image
+        elif page := self.get("link"):
+            model = page.content_type.model_class().__name__
```

Comment:
> **Praise & Suggestion:** This is a nice approach though I wonder whether we could add a doc-string to make this a little clearer for new devs coming on to the project?

Thanks @lparsons396 done in 7fd3455

Reply:
I would not worry about non-work or non-historical work pages, as I am not sure we have suitable generic fields to use for the image on other page types.

Reply:
> **Question:** Even though the page_type is limited in L67, should we add a catch-all to this conditional in case someone extends the page_types to include a non (historical)workpage in future?

I think this should still be covered by:

https://github.com/torchbox/torchbox.com/blob/7fd3455283df3ac57ce7102eb6a2238390b55045/tbx/services/blocks.py#L66